src/main/java/org/shanerx/tradeshop/Utils.java

```diff
@@ -46,219 +51,575 @@
+	}
+	
+	/**
+	 * Returns all the owners of a TradeShop, including the one on the last line of the sign.
+	 * @param b the inventory holder block
+	 * @return all the owners.
+	 */
+	public List<OfflinePlayer> getShopOwners(Block b) {
+        if (!plugin.getAllowedInventories().contains(b.getType())) {
+            return null;
         }
-    }
 
-    public boolean isInfiniteTradeShopSign(Block b) {
-        if (!isSign(b)) {
-            return false;
-        }
-        Sign sign = (Sign) b.getState();
-        if (!ChatColor.stripColor(sign.getLine(0)).equals("[iTrade]")) {
-            return false;
-        } else {
-            return true;
+		List<OfflinePlayer> owners = new ArrayList<>();
+		Inventory inv = ((InventoryHolder) b.getState()).getInventory();
+		String names = inv.getName();
+		for (String m : names.split(";")) {
+			if (m.startsWith("o:")) {
+				owners.add(Bukkit.getOfflinePlayer(m.substring(2)));
+			}
+		}
+		Sign s = findShopSign(b);
+		if (s.getLine(3) == null || s.getLine(3).equals("")) {
+            if (owners.size() > 0) {
+                s.setLine(3, owners.get(0).getName());
+                s.update();
+            }
+            return owners;
+        } else if (!owners.contains(Bukkit.getOfflinePlayer(s.getLine(3)))) {
+			owners.add(Bukkit.getOfflinePlayer(s.getLine(3)));
+            addOwner(s, Bukkit.getOfflinePlayer(s.getLine(3)));
         }
+        return owners;
     }
 
-    public boolean isShopSign(Block b)
-    {
-        return isTradeShopSign(b) || isInfiniteTradeShopSign(b) || isBiTradeShopSign(b);
-    }
-
-    public boolean isSign(Block b)
-    {
-        if(b == null)
-            return false;
-        return b.getType() == Material.SIGN_POST || b.getType() == Material.WALL_SIGN;
-    }
-
-    public boolean isInt(String str) {
-        try {
-            Integer.parseInt(str);
-            return true;
-        } catch (Exception e) {
-            return false;
+    /**
+     * Returns all the owners of a TradeShop, including the one on the last line of the sign.
+     *
+     * @param b the inventory holder block
+     * @return all the owners.
+     */
+    public List<OfflinePlayer> getShopMembers(Block b) {
+        if (!plugin.getAllowedInventories().contains(b.getType())) {
+            return null;
         }
-    }
 
-    public boolean canFit(Inventory inv, ItemStack itm, int amt) {
-        int count = 0, empty = 0;
-        for (ItemStack i : inv.getContents()) {
-            if (i != null) {
-                if (i.getType() == itm.getType() && i.getData() == itm.getData() && i.getDurability() == itm.getDurability() && i.getItemMeta() == itm.getItemMeta()) {
-                    count += i.getAmount();
-                }
+        List<OfflinePlayer> members = new ArrayList<>();
+        Inventory inv = ((InventoryHolder) b.getState()).getInventory();
+        String names = inv.getName();
+        for (String m : names.split(";")) {
+            if (m.startsWith("m:")) {
```

Comment:
I am sure this used to be different. The if chrck should be removed before merging. Another method returning both owners and members is useless.